plugins/modules/aci_fabric_external_connection_profile.py

```diff
@@ -0,0 +1,305 @@
+        "text": "unknown managed object class foo"
+    }
+raw:
+  description: The raw output returned by the APIC REST API (xml or json)
+  returned: parse error
+  type: str
+  sample: '<?xml version="1.0" encoding="UTF-8"?><imdata totalCount="1"><error code="122" text="unknown managed object class foo"/></imdata>'
+sent:
+  description: The actual/minimal configuration pushed to the APIC
+  returned: info
+  type: list
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment"
+            }
+        }
+    }
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        description=dict(type="str", aliases=["descr"]),
+        fabric_id=dict(type="int", aliases=["id", "fabric"]),
+        name=dict(type="str", aliases=["profile_name"]),
+        community=dict(type="str", aliases=["rt", "route_target"]),
+        site_id=dict(type="int", aliases=["sid", "site", "s_id"]),
+        peering_type=dict(
+            type="str", default="automatic_with_full_mesh", aliases=["p_type", "peer", "peer_t"], choices=["automatic_with_full_mesh", "automatic_with_rr"]
```

Comment:
We typically do not set defaults in the modules. I think we should remove default, and explain in the comment that defaults to automatic_with_full_mesh when not provided.

```suggestion
            type="str", aliases=["p_type", "peer", "peer_t"], choices=["automatic_with_full_mesh", "automatic_with_rr"]
```

Reply:
Yes I am aware of that. However, as I mentioned in this previous comment -
> I did have to change the option to include a default for peering_type because not setting the type doesn't default in the APIC which appears to cause invalid config.
https://github.com/CiscoDevNet/ansible-aci/pull/630#discussion_r1518469224

Reply:
Did short test, but I did not have to provide any of the config when changing the description. I can also set/reset password only, when not providing the type. I think we should construct the child payload based on provided input only. This way default should not be specified and you are not sending None to apic. Also this allows for just providing password in case you want to do a password change with rr_type.

Reply:
Okay have fixed it and also added an extra test case.